torch_xla/experimental/xla_sharded_tensor.py

```diff
@@ -48,6 +48,20 @@ class XLAShardedTensor(torch.Tensor):
   # Note: we can consider returning a callback or even define
   # sharding at XLAShardedTensor construction after pjrt migration.
   local_shards: List[XLAShard] = None
+  # A logical device topology, each element describes
+  # a number of devices in the corresponding axis.
+  # NOTE: we could use more specific device-rank mapping, e.g., ShardingSpec,
+  # if needed. The change shouldn't be difficult, or create another constructor.
+  mesh_shape: Tuple[int]  # TODO: create a wrapper for named axes
```

Comment:
Regarding your TODO, this might be helpful: https://github.com/pytorch/xla/blob/b7f54864a0933d6401a023e12ec6ccac1982e2d9/torch_xla/experimental/tpu.py#L32-L51

Reply:
+1 thanks :)